packages/core/content-manager/server/src/history/services/utils.ts

```diff
@@ -352,6 +351,46 @@ export const createServiceUtils = ({ strapi }: { strapi: Core.Strapi }) => {
     );
   };
 
+  /**
+   * @description
+   * Builds a key path to the next nested component (fieldA.component.fieldB.componentB...)
+   * @returns
+   * The updated key path and the next component's schema
+   */
+  const getNextComponent = (
+    componentKeyPath: string[],
+    componentName: string,
+    componentSchema: Schema.Attribute.AnyAttribute
+  ) => {
+    if (componentSchema.type !== 'component') return;
+
+    // Update the keyPath to include a component that has never been visited
+    const keyPath = [...componentKeyPath];
+    if (!keyPath.includes(componentName)) {
```

Comment:
Can components be nested inside themselves?

Reply:
Anytime I've tried the app has crashed...

Reply:
all good then